File: 2021/06/14/main.go
package main

import "sort"

func maximumUnits(boxTypes [][]int, truckSize int) int {
	sort.Slice(boxTypes, func(i, j int) bool {
		return boxTypes[i][1] > boxTypes[j][1]
	})
	cnt, res := 0, 0
	for i := 0; i < len(boxTypes); i++ {
		for j := 0; j < boxTypes[i][0]; j++ {
			if cnt < truckSize {
				res += boxTypes[i][1]
				cnt += 1
			}
		}
	}
	return res
}

File: 2021/06/14/main.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1], reverse=True)
        cnt = 0
        res = 0
        for box in boxTypes:
            for num in range(box[0]):
                if cnt < truckSize:
                    res += box[1]
                    cnt += 1
        return res